core/domain/user_jobs_one_off_test.py

```diff
@@ -377,3 +385,443 @@ def test_adding_exploration_to_collection(self):
+                user_jobs_one_off.UserImpactCalculationOneOffJob.enqueue(job_id)
+                self.process_and_flush_pending_tasks()
+
+    def _run_exp_impact_calculation_and_assert_equals(
+            self, exploration_id, expected_impact):
+        with self.swap(stats_jobs_continuous.StatisticsAggregator,
+                    'get_statistics', self._mock_get_statistics):
+            self.assertEqual(expected_impact,
+                user_jobs_one_off.UserImpactCalculationOneOffJob._get_exp_impact_score(
+                exploration_id))
+
+    def _sign_up_user(self, user_email, username):
+        # Sign up a user, have them create an exploration.
+        self.signup(user_email, username)
+        return self.get_user_id_from_email(user_email)
+
+    def _create_exploration(self, exp_id, user_id):
+        exploration = exp_domain.Exploration.create_default_exploration(
+            exp_id, 'A title', 'A category')
+        exp_services.save_new_exploration(user_id, exploration)
+        return exploration
+
+    def _rate_exploration(self, exp_id, num_ratings, rating):
+        """Create num_ratings ratings for exploration with exp_id,
+        of value rating.
+        """
+        # Each user id needs to be unique since each user can only give an
+        # exploration one rating.
+        user_ids = ['user{}'.format(i) for i in range(num_ratings)]
+        for user_id in user_ids:
+            rating_services.assign_rating_to_exploration(
+                user_id, exp_id, rating
+            )
+
+    def _complete_exploration(self, exploration, num_completions):
+        """Log a completion of exploration with id exp_id num_completions
+        times."""
+        exp_version = 1
+        state = exploration.init_state_name
+        session_ids = ['session{}'.format(i) for i in range(num_completions)]
+        for session_id in session_ids:
+            event_services.StartExplorationEventHandler.record(
+                exploration.id, exp_version, state, session_id, {},
+                feconf.PLAY_TYPE_NORMAL)
+            event_services.CompleteExplorationEventHandler.record(
+                exploration.id, exp_version, state, session_id, 27, {},
+                feconf.PLAY_TYPE_NORMAL)
+        # Set the number of completions, so mock can function
+        # correctly.
+        self.num_completions[exploration.id] = num_completions
+
+    def test_user_with_no_explorations_has_no_impact(self):
+        """Test that a user who is not a contributor on any exploration
+        is not assigned an impact score by the UserImpactCalculationOneOffJob.
+        """
+        self.user_a_id = self._sign_up_user(
+            self.USER_A_EMAIL, self.USER_A_USERNAME)
+        self._run_one_off_job()
+        user_stats_model = user_models.UserStatsModel.get(
+            self.user_a_id, strict=False)
+        self.assertIsNone(user_stats_model)
+
+    def test_standard_user_impact_calculation_one_exploration(self):
```

Comment:
I think it would be good to test that you get the expected impact with different ratings, number of completions etc.